sonic_data_client/virtual_db.go

```diff
@@ -284,6 +301,85 @@ func getCountersMap(tableName string) (map[string]string, error) {
 	return counter_map, nil
 }
 
+
+// Get the mapping between objects in counters DB, Ex. port name to oid in "COUNTERS_PORT_NAME_MAP" table.
+// Aussuming static port name to oid map in COUNTERS table
+func getFabricCountersMap(tableName string) (map[string]string, error) {
+	counter_map := make(map[string]string)
+	dbName := "COUNTERS_DB"
+	redis_client_map, err := GetRedisClientsForDb(dbName)
+	if err != nil {
+                return nil, err
+	}
+	for namespace, redisDb := range redis_client_map {
+		fv, err := redisDb.HGetAll(tableName).Result()
+		if err != nil {
+			log.V(2).Infof("redis HGetAll failed for COUNTERS_DB in namespace %v, tableName: %s", namespace, tableName)
+			return nil, err
+		}
+		namespaceFv := make(map[string]string)
+		for k, v := range fv {
+			var namespace_str = ""
+            if len(namespace) != 0 {
+                namespace_str = string('-') + namespace
+            }
+			namespaceFv[k + namespace_str] = v
+		}
+		addmap(counter_map, namespaceFv)
+		log.V(6).Infof("tableName: %s in namespace %v, map %v", tableName, namespace, namespaceFv)
+	}
+	return counter_map, nil
+}
+
+// Populate real data paths from paths like
+// [COUNTER_DB COUNTERS Ethernet*] or [COUNTER_DB COUNTERS Ethernet68]
+func v2rFabricPortStats(paths []string) ([]tablePath, error) {
+	var tblPaths []tablePath
+	fmt.Errorf("in v2rFabricPortStats")
+	if strings.HasSuffix(paths[KeyIdx], "*") { // All Ethernet ports
+		for port, oid := range countersFabricPortNameMap {
+			var namespace string
+			if strings.Contains(port, "-"){
+			    namespace = strings.Split(port, "-")[1]
+			} else {
+				namespace = ""
+			}
+			separator, _ := GetTableKeySeparator(paths[DbIdx], namespace)
+			tblPath := tablePath{
+				dbNamespace:  namespace,
+				dbName:       paths[DbIdx],
+				tableName:    paths[TblIdx],
+				tableKey:     oid,
+				delimitor:    separator,
```

Comment:
can we add UT for this scenario ?

Reply:
Added unit-test to check testGnmiGet for single and multi-namespace scenarios.
This checks if PORT0-asic0 is present for multi-ns and PORT0 is present for single asic.
Also checks for COUNTERS/PORT*